pkg/auth/auth.go

```diff
@@ -43,6 +44,41 @@ func CheckAuthFile(authfile string) error {
 	return nil
 }
 
+// Mirrors --authfile to a tmpfile if --authfile points to a
+// file descriptor instead of actual file on filesystem
+// reason is file descriptor could vanish mid-session if consumed
+// once, check https://github.com/containers/buildah/issues/3070
```

Comment:
Pointing to an issue of another projects isn't a very helpful description of an API.  Rather put inside the function body.

We need to make sure that the APIs is self-contained, so flags (e.g., --authfile) won't add much information either.

Reading the function, I am not sure the logic is necessarily specific to authfiles.  It seems rather something for a utils-like package and be generalized for any kind of file?